onnxruntime/core/optimizer/cast_chain_elimination.cc

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/common/logging/logging.h"
+#include "core/optimizer/cast_chain_elimination.h"
+#include "core/optimizer/utils.h"
+#include "core/graph/graph.h"
+#include "core/graph/graph_utils.h"
+
+namespace onnxruntime {
+
+Status CastChainElimination::ApplyImpl(Graph& graph, bool& modified, int graph_level, const logging::Logger& logger) const {
+  GraphViewer graph_viewer(graph);
+  const auto& node_topology_list = graph_viewer.GetNodesInTopologicalOrder();
+
+  for (auto node_index : node_topology_list) {
+    auto* node_ptr = graph.GetNode(node_index);
+    if (node_ptr == nullptr)
+      continue;  // we removed the node as part of an earlier fusion
+
+    Node& node = *node_ptr;
+
+    ORT_RETURN_IF_ERROR(Recurse(node, modified, graph_level, logger));
+
+    if (!graph_utils::CanRemoveNode(graph, node, logger)) {
+      continue;
+    }
+
+    if (node.OpType() != "Cast") {
+      continue;
+    }
+
+    const auto* input_type = node.InputDefs()[0]->TypeAsProto();
+    if (input_type == nullptr || !input_type->tensor_type().has_elem_type()) {
+      continue;
+    }
+
+    // If not, find the longest chain that casts to the input type, if it exists.
+    Node* current = &node;
+    Node* final_non_cast_node = &node;
+    int matching_elem_type = input_type->tensor_type().elem_type();
+
+    while (current->OpType() == "Cast") {
+      const auto& to_attr = current->GetAttributes().at("to");
```

Comment:
move this line after checks (like line 55).

It is because, if code is returned or break loop, there is no need to read the to attribute.